test/merge-patient-lab-iips-mocha.js

```diff
@@ -5,151 +5,260 @@ var chaiAsPromised = require('chai-as-promised');
+
+                            // initialize state
+                            component.rpf.vni('').inputState('patient',
+                                                             { id: '001', 
+                                                               name: 'Alice', 
+                                                               dob: '1979-01-23' });
+
+                            var state = component.rpf.vni('').inputState('patient');
+                            state.should.be.an('object');
+                            state.should.have.all.keys( 'data', 'lm', 'previousLms', 'port' );
+                    });
+                });
+
+                it("should have input state data after input", function() {
+                    return Promise.resolve(componentFactory.getComponent )
+                        .then(commonTest.createComponent).then(function(component){
+
+                            // initialize state
+                            var patientState = { id: '001', name: 'Alice', dob: '1979-01-23' };
+                            component.rpf.vni('').inputState('patient',
+                                                             patientState );
+
+                            var state = component.rpf.vni('').inputState('patient');
+                            state.should.be.an('object');
+			    state.data.should.deep.equal( patientState );
+                    });
+                });
             });
-        });
-        describe('#getOutputState', function() {
-            it("should initially return nothing", function() {
-                return Promise.resolve(createComponent()).then(function(component){
-                    return component.rpf.createVni('').getOutputState();
-                }).should.become(undefined);
+
+            describe('#outputState', function() {
+
+                it("should initially return nothing", function() {
+                    return Promise.resolve(componentFactory.getComponent )
+                        .then(commonTest.createComponent).then(function(component){
+                            return component.rpf.vni('').outputState();
+                    }).should.become(undefined);
+                });
+
+                it("should have a output state after input", function() {
+                    return Promise.resolve(componentFactory.getComponent )
+                        .then(commonTest.createComponent).then(function(component){
+
+                            // initialize state
+                            component.rpf.vni('').outputState( { id: '001', 
+                                                                 name: 'Alice', 
+                                                                 dob: '1979-01-23', 
+                                                                 glucose: '75',  
+                                                                 date: '2012-02-01'});
```

Comment:
The outputState function should be a getter/setter for an *entire* state object -- not just the data property of that object.  It should not automatically set the lm property.  Generation of a new LM should be handled at a higher level in the Framework, because the lm should not always be changed just because the data property is changed.  Although it would be rare, a node should be able to update its state.data without updating its state.lm .